### ISSUE TITLE
CSRF token support in image upload

### ISSUE THREAD

**Issue by kentwait** [greyli/flask-ckeditor#5]
If CSRF protection is enabled in Flask via `CSRFProtect(app)`, clicking "Send it to the server" in the image upload tab will return an error an error - HTTP 400  "CSRF token missing"

**Comment by greyli** [comment]
CKEditor use AJAX to send file, so you can use the code below to configure AJAX requests. This will add the CSRF token in `X-CSRFToken` header:
```javascript
<script type="text/javascript">
    var csrf_token = "{{ csrf_token() }}";

    $.ajaxSetup({
        beforeSend: function(xhr, settings) {
            if (!/^(GET|HEAD|OPTIONS|TRACE)$/i.test(settings.type) && !this.crossDomain) {
                xhr.setRequestHeader("X-CSRFToken", csrf_token);
            }
        }
    });
</script>
```

**Comment by greyli** [comment]
Implemented in [0.4.3](https://github.com/greyli/flask-ckeditor/releases/tag/0.4.3). Just set `CKEDITOR_ENABLE_CSRF` to `True`.

**Comment by greyli** [comment]
I’ll close this issue for now. Please feel free to comment further or re-open this issue if you wish.
